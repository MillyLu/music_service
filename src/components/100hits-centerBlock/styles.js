import styled from "styled-components";

export const MainCenterblock = styled.div`
width: auto;
flex-grow: 3;
padding: 20px 40px 20px 111px;
background-color: ${props => props.theme === "light" ? "#FFFFFF" : "#1C1C1C"}; 
`;

export const CenterblockContent = styled.div`
display: flex;
flex-direction: column;
width: 1107px;
`;