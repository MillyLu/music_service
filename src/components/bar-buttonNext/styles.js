import styled from "styled-components";
import {ReactComponent as NextSvg} from '../svg/next/next.svg';

export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonNext = styled(PlayerButton)`
margin-right: 28px;
fill: #a53939;
`;

export const PlayerButtonNextSvg = styled(NextSvg)`
width: 15px;
height: 14px;
fill: ${props => props.theme === "light" ? "#B1B1B1" : "#D9D9D9"};
 & :hover {
    fill: ${props => props.theme === "light" ? "#707070" : "#696969"};
 }
`;