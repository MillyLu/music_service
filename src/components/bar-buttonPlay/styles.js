import styled from "styled-components";

export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const ButtonCursor = styled(PlayerButton)`
cursor: pointer;
`

export const PlayerButtonPlay = styled(ButtonCursor)`
margin-right: 23px;
`;

export const PlayerButtonPlaySvg = styled.svg`
width: 22px;
  height: 20px;
  fill: ${props => props.theme === "light" ? "#B1B1B1" : "#D9D9D9"};
  & :hover {
     fill: ${props => props.theme === "light" ? "#707070" : "#696969"};
  }
`;

export const Audio = styled.audio`
display: none;
`