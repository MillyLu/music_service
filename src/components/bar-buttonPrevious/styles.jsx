import styled from "styled-components";
import {ReactComponent as PrevSvg} from '../svg/prev/prev.svg';

export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonPrev = styled(PlayerButton)`
margin-right: 23px;
`;

export const PlayerButtonPrevSvg = styled(PrevSvg)`
width: 15px;
height: 14px;
fill: ${props => props.theme === "light" ? "#B1B1B1" : "#D9D9D9"};
 & :hover {
    fill: ${props => props.theme === "light" ? "#707070" : "#696969"};
 }
`;