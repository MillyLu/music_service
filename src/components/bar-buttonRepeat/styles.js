import styled from "styled-components";
import {ReactComponent as RepeatSvg} from '../svg/repeat/repeat.svg';


export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonIcon = styled(PlayerButton)``;

export const PlayerButtonRepeat = styled(PlayerButtonIcon)`
margin-right: 24px;
`;

export const PlayerButtonRepeatSvg = styled(RepeatSvg)`
width: 18px;
height: 12px;
fill: none;
stroke: ${props => props.theme === "light" ? "#B1B1B1" : "#696969"};

${PlayerButtonRepeat}:hover & {
    fill: none;
    stroke: ${props => props.theme === "light" ? "#707070" : "#ACACAC"};
    cursor: pointer;
  }
${PlayerButtonRepeat}:active & {
    fill: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
    stroke: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
    cursor: pointer;
  }
`;