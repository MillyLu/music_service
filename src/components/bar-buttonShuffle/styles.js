import styled from "styled-components";
import {ReactComponent as ShuffleSvg} from '../svg/shuffle/shuffle.svg';

export const PlayerButton = styled.div`
padding: 5px;
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonIcon = styled(PlayerButton)``;

export const PlayerButtonShuffle = styled(PlayerButtonIcon)`
display: -webkit-box;
display: -ms-flexbox;
display: flex;
-webkit-box-align: center;
    -ms-flex-align: center;
        align-items: center;
`;

export const PlayerButtonShuffleSvg = styled(ShuffleSvg)`
width: 19px;
height: 12px;
fill: none;
stroke: ${props => props.theme === "light" ? "#B1B1B1" : "#696969"};
${PlayerButtonShuffle}:hover & {
    fill: none;
    stroke: ${props => props.theme === "light" ? "#707070" : "#ACACAC"};
    cursor: pointer;
  }
${PlayerButtonShuffle}:active & {
    fill: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
    stroke: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
    cursor: pointer;
  }
`;