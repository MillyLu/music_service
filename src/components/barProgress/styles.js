import styled from "styled-components";


export const BarPlayerProgress = styled.div`
width: 100%;
height: 5px;
background: #2E2E2E;
`;

export const Progress = styled.input`
width: 100%;
height: 5px;
vertical-align: top;
`
