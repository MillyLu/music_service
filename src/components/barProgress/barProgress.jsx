/* eslint-disable no-param-reassign */
import * as Styled from './styles';


export function BarProgress({
    inputRef,
    aud

  }) {

    const handleProgressChange = () => {
        aud.current.currentTime = inputRef.current.value;
      };


    return (
        <Styled.BarPlayerProgress>
             
              <Styled.Progress data-testid="progress" type="range" defaultValue='0' ref={inputRef} onChange={handleProgressChange}></Styled.Progress>

        </Styled.BarPlayerProgress>
    )
}
