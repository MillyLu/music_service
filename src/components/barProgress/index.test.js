/* eslint-disable no-undef */
import { screen, render,  fireEvent } from "@testing-library/react";
// eslint-disable-next-line import/no-unresolved

import { BarProgress } from "./barProgress";
import * as Styled from './styles';



describe("<BarProgress />", () => {
  describe("Snapshots tests", () => {
    it("should render successfully and match snapshot", () => {
      const { container } = render(<BarProgress></BarProgress>);

      expect(container).toMatchSnapshot();
    });
  });
});


  test('It should change onChange', () => {
    const handleProgressChange = jest.fn();
    render(<Styled.Progress type="range" data-testid="progress" defaultValue='0' onChange={handleProgressChange}></Styled.Progress>)
    const input = screen.getByTestId("progress")
    fireEvent.change(input, {target: {value: '60'}})
    expect(handleProgressChange).toHaveBeenCalledTimes(1)
    expect(input.value).toBe('60')
  })


