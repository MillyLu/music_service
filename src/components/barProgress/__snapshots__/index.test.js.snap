// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarProgress /> Snapshots tests should render successfully and match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe kDyUME"
  >
    <input
      class="sc-gswNZR fvGZDH"
      data-testid="progress"
      type="range"
      value="0"
    />
  </div>
</div>
`;
